plugins/modules/zos_job_submit.py

```diff
@@ -1039,7 +1036,7 @@ def run_module():
         module.exit_json(**result)
 
     finally:
-        if temp_file:
+        if location != "DATA_SET" and location != "USS":
```

Comment:
```suggestion
        if temp_file:
```

I suggest keeping it this way bc that will help us avoid having to check for state and just remove it if it exists.

Reply:
It generates a error if the local file is named as a dataset (USER.LOCAL) then the file /user.local will fail that verification

Reply:
I'm sorry I don't get how that relates to this code suggestion, could you explain?